source/common/network/utility.cc

```diff
@@ -542,11 +539,71 @@ void passPayloadToProcessor(uint64_t bytes_read, Buffer::RawSlice& slice,
                                      std::move(buffer), receive_time);
 }
 
+Api::IoCallUint64Result receiveMessage(uint64_t max_packet_size, Buffer::InstancePtr& buffer,
+                                       Buffer::RawSlice& slice, IoHandle::RecvMsgOutput& output,
```

Comment:
Do we really need to pass in slice reference?

Reply:
`slice.len_` is being used later to iterate over the gso_sized sub_buffers. 
Hence we pass slice by reference here, so that we receive the updated value of `slice.len_` later.

Reply:
nit: use buffer.length() there and remove `slice` from the parameter list.

Reply:
Thanks for the suggestion! 

Removed slice parameter from the arguments for the receiveMessage function
Using buffer->length() to iterate over the buffer during gro_segmentation.